starter-code/src/functions-and-arrays.js

```diff
@@ -11,14 +20,43 @@ var words = [
   'crackpot'
 ];
 
+function findLongestWord (array) {
+  var longest = 0; 
+  for (var i = 0; i < array.length; i++) {
+    if (array[i].length > longest) {
+      longest = array[i].length;
+    } 
+      return longest;
+    }
+  } 
+  
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(array) {
+  if (array !== "") {
```

Comment:
Os falla que no estáis retornando nada cuando hacéis el reduce, os faltaría después del else: `return result`. y otro detalle, cuando estáis utilizando la propiedad reduce tenéis que darle un valor inicial. En total se os quedaría un código así.

```
  if (array !== "") {
  var result=array.reduce(function(a,b){
  return a + b;  
  },0)
} else {
  return 0;
}
return result;
```